File: optimize-partner-logos.py
#!/usr/bin/env python3
"""
Script to optimize partner/B2B logos
Resizes and compresses logos for web use
"""

import os
import sys
from pathlib import Path
from PIL import Image

def optimize_logo(input_path, max_dimension=400, quality=85):
    """
    Optimize a logo image
    
    Args:
        input_path: Path to the input image
        max_dimension: Maximum width or height (default 400px)
        quality: JPEG quality (1-100, default 85)
    
    Returns:
        Path to optimized image or None if failed
    """
    try:
        img = Image.open(input_path)
        original_size = os.path.getsize(input_path)
        
        print(f"\n📸 Optimizing: {os.path.basename(input_path)}")
        print(f"   Original: {img.size[0]}x{img.size[1]}, {original_size / 1024:.2f} KB")
        
        # Calculate new dimensions
        width, height = img.size
        if width > max_dimension or height > max_dimension:
            if width > height:
                new_width = max_dimension
                new_height = int((height / width) * max_dimension)
            else:
                new_height = max_dimension
                new_width = int((width / height) * max_dimension)
        else:
            new_width, new_height = width, height
        
        # Convert to RGB if necessary (for JPEG)
        if img.mode in ('RGBA', 'LA', 'P'):
            # Create white background
            background = Image.new('RGB', (new_width, new_height), (255, 255, 255))
            if img.mode == 'P':
                img = img.convert('RGBA')
            resized = img.resize((new_width, new_height), Image.Resampling.LANCZOS)
            if resized.mode in ('RGBA', 'LA'):
                background.paste(resized, mask=resized.split()[-1] if resized.mode in ('RGBA', 'LA') else None)
            else:
                background.paste(resized)
            img = background
        elif img.mode != 'RGB':
            img = img.convert('RGB')
            img = img.resize((new_width, new_height), Image.Resampling.LANCZOS)
        else:
            img = img.resize((new_width, new_height), Image.Resampling.LANCZOS)
        
        # Determine output format
        ext = Path(input_path).suffix.lower()
        output_path = input_path.replace(ext, '-optimized.jpg' if ext in ['.jpeg', '.jpg'] else '-optimized.png')
        
        # Save optimized image
        if ext in ['.jpeg', '.jpg']:
            img.save(output_path, 'JPEG', quality=quality, optimize=True)
        elif ext == '.png':
            img.save(output_path, 'PNG', optimize=True, compress_level=9)
        elif ext == '.gif':
            # For GIF, convert to PNG for better optimization
            output_path = input_path.replace('.gif', '-optimized.png')
            img.save(output_path, 'PNG', optimize=True, compress_level=9)
            print(f"   ⚠️  GIF converted to PNG for better optimization")
        else:
            print(f"   ⚠️  Unsupported format: {ext}")
            return None
        
        optimized_size = os.path.getsize(output_path)
        savings = ((1 - optimized_size / original_size) * 100) if original_size > 0 else 0
        
        print(f"   ✅ Optimized: {new_width}x{new_height}, {optimized_size / 1024:.2f} KB ({savings:.1f}% smaller)")
        
        return output_path
    except Exception as e:
        print(f"   ❌ Error: {str(e)}")
        return None

def main():
    script_dir = Path(__file__).parent
    partners_dir = script_dir / "public" / "images" / "partners"
    
    if not partners_dir.exists():
        print(f"❌ Partners directory not found: {partners_dir}")
        sys.exit(1)
    
    # Find all image files
    image_extensions = ['.jpg', '.jpeg', '.png', '.gif']
    files = [f for f in partners_dir.iterdir() 
             if f.is_file() and f.suffix.lower() in image_extensions 
             and '-optimized' not in f.name]
    
    if not files:
        print("ℹ️  No image files found in partners directory")
        return
    
    print(f"🚀 Starting logo optimization...\n")
    print(f"📂 Found {len(files)} logo file(s) to optimize:\n")
    for f in files:
        print(f"   - {f.name}")
    
    optimized_count = 0
    for file_path in files:
        result = optimize_logo(str(file_path))
        if result:
            optimized_count += 1
    
    print(f"\n{'='*60}")
    print(f"📊 Summary:")
    print(f"   ✅ Optimized: {optimized_count}/{len(files)} files")
    print(f"{'='*60}\n")
    print("💡 Tip: Review optimized files and replace originals if satisfied.")
    print("   You can rename optimized files to replace originals.")

if __name__ == "__main__":
    main()

File: optimize-partner-logos.js
#!/usr/bin/env node
/**
 * Script to optimize partner/B2B logos
 * Resizes and compresses logos for web use
 */

import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const partnersDir = path.join(__dirname, 'public/images/partners');

// Check if sharp is available for optimization
let sharp;
try {
  sharp = (await import('sharp')).default;
} catch (e) {
  console.log('⚠️  Sharp not available. Using basic optimization...');
}

async function optimizeImage(filePath) {
  const ext = path.extname(filePath).toLowerCase();
  const basename = path.basename(filePath, ext);
  const dir = path.dirname(filePath);
  
  try {
    if (sharp) {
      const image = sharp(filePath);
      const metadata = await image.metadata();
      
      console.log(`\n📸 Optimizing: ${path.basename(filePath)}`);
      console.log(`   Original: ${metadata.width}x${metadata.height}, ${(fs.statSync(filePath).size / 1024).toFixed(2)} KB`);
      
      // For logos, we typically want max 400px width/height
      const maxDimension = 400;
      let width = metadata.width;
      let height = metadata.height;
      
      if (width > maxDimension || height > maxDimension) {
        if (width > height) {
          width = maxDimension;
          height = Math.round((metadata.height / metadata.width) * maxDimension);
        } else {
          height = maxDimension;
          width = Math.round((metadata.width / metadata.height) * maxDimension);
        }
      }
      
      // Optimize based on format
      if (ext === '.jpeg' || ext === '.jpg') {
        await image
          .resize(width, height, { fit: 'inside', withoutEnlargement: true })
          .jpeg({ quality: 85, progressive: true, mozjpeg: true })
          .toFile(path.join(dir, `${basename}-optimized.jpg`));
      } else if (ext === '.png') {
        await image
          .resize(width, height, { fit: 'inside', withoutEnlargement: true })
          .png({ quality: 90, compressionLevel: 9 })
          .toFile(path.join(dir, `${basename}-optimized.png`));
      } else if (ext === '.gif') {
        // For GIF, we'll keep original but can create a PNG version
        console.log(`   ⚠️  GIF format - keeping original. Consider converting to PNG for better optimization.`);
        return filePath;
      }
      
      const optimizedPath = path.join(dir, `${basename}-optimized${ext}`);
      const originalSize = fs.statSync(filePath).size;
      const optimizedSize = fs.statSync(optimizedPath).size;
      const savings = ((1 - optimizedSize / originalSize) * 100).toFixed(1);
      
      console.log(`   ✅ Optimized: ${width}x${height}, ${(optimizedSize / 1024).toFixed(2)} KB (${savings}% smaller)`);
      
      return optimizedPath;
    } else {
      console.log(`⚠️  Sharp not available. Skipping optimization for: ${path.basename(filePath)}`);
      return filePath;
    }
  } catch (error) {
    console.error(`❌ Error optimizing ${path.basename(filePath)}:`, error.message);
    return filePath;
  }
}

async function main() {
  console.log('🚀 Starting logo optimization...\n');
  
  if (!fs.existsSync(partnersDir)) {
    console.error(`❌ Partners directory not found: ${partnersDir}`);
    process.exit(1);
  }
  
  const files = fs.readdirSync(partnersDir).filter(file => {
    const ext = path.extname(file).toLowerCase();
    return ['.jpg', '.jpeg', '.png', '.gif', '.webp'].includes(ext);
  });
  
  if (files.length === 0) {
    console.log('ℹ️  No image files found in partners directory');
    return;
  }
  
  console.log(`📂 Found ${files.length} logo file(s) to optimize:\n`);
  files.forEach(file => console.log(`   - ${file}`));
  
  for (const file of files) {
    const filePath = path.join(partnersDir, file);
    await optimizeImage(filePath);
  }
  
  console.log('\n✨ Optimization complete!\n');
  console.log('💡 Tip: Review optimized files and replace originals if satisfied.');
}

main().catch(console.error);










